pandas/core/indexes/base.py

```diff
@@ -4636,8 +4636,9 @@ def join(
             return self._join_non_unique(other, how=how)
         elif not self.is_unique or not other.is_unique:
             if self.is_monotonic_increasing and other.is_monotonic_increasing:
-                if self._can_use_libjoin:
+                if not is_interval_dtype(self.dtype):
```

Comment:
So `_can_use_libjoin` shouldn't be updated instead?

Reply:
No, in most cases going through object dtype is slower than the alternative. In our case here the alternative can not handle non interval dtype